src/IO/HDF5/HDF5IOHandler.cpp

```diff
@@ -352,6 +352,11 @@ HDF5IOHandlerImpl::openFile(Writable* writable,
     if( !auxiliary::ends_with(name, ".h5") )
         name += ".h5";
 
+    auto search = m_fileIDs.find(writable);
```

Comment:
This is effectively implementing the `//TODO` above, isn't it?
Maybe we can add this as a comment before this block?

Reply:
Yes,  looks like so.  Well  CI failed.  let me address that first